aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -409,20 +414,34 @@ private String getIdentityValue(String identityClaim) throws ApiException {
         return groups;
     }
 
-    private CognitoUserPoolsAuthProvider getAuthProvider() throws ApiException {
-        CognitoUserPoolsAuthProvider cognitoProvider = authProvider.getCognitoUserPoolsAuthProvider();
-        if (cognitoProvider == null) {
-            try {
-                cognitoProvider = new DefaultCognitoUserPoolsAuthProvider();
-            } catch (ApiException exception) {
+    private String getAuthToken(AuthorizationType authType) throws ApiException {
```

Comment:
The last few PRs for auth have gone right into the main body of this `AWSApiPlugin`. I'm worried about this. We should really be breaking the auth logic out into smaller classes.

Plugins, in a perfect world, are simply adapters between our business logic and the Amplify framework.

However, increasingly, this `AWSApiPlugin` is host to business logic.

@raphkim The model you used for your Predictions category -- where you had business logic services, and used the plugin as a thin-veneer into them -- was architecturally correct, for example.

Reply:
to be addressed in a separate PR